modules/virt-creating-a-service-from-a-virtual-machine.adoc

```diff
@@ -112,148 +48,74 @@ metadata:
   name: vmservice <1>
   namespace: example-namespace <2>
 spec:
+  externalTrafficPolicy: Cluster <3>
   ports:
-  - port: 27017
+  - nodePort: 30000 <4>
+    port: 27017
     protocol: TCP
-    targetPort: 22 <3>
+    targetPort: 22 <5>
   selector:
-    special: key <4>
-  type: ClusterIP <5>
+    special: key <6>
+  type: NodePort <7>
 ----
-<1> Specify the `name` of the service you are creating and exposing.
-<2> Specify `namespace` in the `metadata` section of the `Service` YAML that corresponds to the `namespace` you specify in the virtual machine YAML.
-<3> Add `targetPort: 22`, exposing the service on SSH port `22`.
-<4> In the `spec` section of the `Service` YAML, add `special: key` to the `selector` attribute, which corresponds to the `labels` you added in the virtual machine YAML configuration file.
-<5> In the `spec` section of the `Service` YAML, add `type: ClusterIP` for a
-ClusterIP service. To create and expose other types of services externally, outside of the cluster, such as `NodePort` and `LoadBalancer`, replace
-`type: ClusterIP` with `type: NodePort` or `type: LoadBalancer`, as appropriate.
-+
+<1> The name of the `Service` object.
+<2> The namespace in which the `Service` object resides. This must match the `metadata.namespace` field of the `VirtualMachine` manifest.
+<3> Optional: Specifies how the nodes distribute service traffic that is received on external IP addresses. This only applies to `NodePort` and `LoadBalancer` service types. The default value is `Cluster` which routes traffic to all endpoints evenly.
```

Comment:
I had to read the kubernetes docs to figure out what "routes traffic to all endpoints evenly" meant; the [docs](https://kubernetes.io/docs/concepts/services-networking/service/#external-traffic-policy) say:
> Set the field to Cluster to route external traffic to all ready endpoints and Local to only route to ready node-local endpoints. If the traffic policy is Local and there are no node-local endpoints, the kube-proxy does not forward any traffic for the relevant Service.

I think the following suggestion would be helpful:
```suggestion
<3> Optional: Specifies how the nodes distribute service traffic that is received on external IP addresses. This only applies to `NodePort` and `LoadBalancer` service types. The default value is `Cluster` which routes traffic to all endpoints across the cluster evenly.
```